docs/specification.md

```diff
@@ -0,0 +1,1264 @@
+        (address %owner)
+        (pair
+          (address %operator)
+          (or %tokens
+            (unit %allTokens)
+            (set %someTokens nat)
+          )
+      ))
+      (bool %isOperator)
+    )
+  )
+)
+```
+
+- Inspect if an address is an operator for the specified owner and 
+  token types.
+
+- If the adddress is not an operator for at least one
+  requested token type, then the result is `false`.
+
+- It's possible to make this query for some specific tokens, or to
+  all tokens.
+
+**pause**
+
+Types
+```
+pause = (address :sender)
+```
+
+Parameter (in Michelson)
+```
+(pair %pause
+  (address %sender)
+  unit
+)
+```
+
+- Pauses transferring, burning and minting operations so
+  that they cannot be performed. All other operations remain
+  unaffected.
+
+- Cannot be called multiple times.
+
+- Sender must be a pauser.
+
+**unpause**
+
+Types
+```
+unpause = (address :sender)
+```
+
+Parameter (in Michelson)
+```
+(pair %unpause
+  (address %sender)
+  unit
+)
+```
+
+- Unpauses the contract so that transferring, burning and minting
+  operations can be performed by users with corresponding roles.
+
+- Cannot be called multiple times.
+
+- Sender must be a pauser.
+
+**getPaused**
+
+Types
+```
+tokenId = nat
+
+getPausedResponse =
+  ( tokenId :tokenId
+  , nat     :paused
+  )
+
+getPausedParam =
+  ( list tokenId                      :tokenIds
+  , contract (list getPausedResponse) :callback
+  )
+
+getPaused = getPausedParam
+```
+
+Parameter (in Michelson)
+```
+(pair %getTotalPaused
+  (list %tokenIds nat)
+  (contract %callback
+    (list
+      (pair %getPausedResponse
+        (nat %tokenId)
+        (bool %paused)
+      )
+    )
+  )
+)
+```
+
+- Returns token pause status.
+
+- *Note: It's not a part of ManagedLedger interface*.
+
+**Custom token functions**
+==========================
+
+**Issuing and destroyng tokens**
+--------------------------
+
+**configureMinter**
+
+Types
+```
+tokenId = nat
+
+configureMinterParam =
+  ( address :sender
+  , address :minter
+  , tokenId :tokenId
+  )
+
+configureMinter = configureMinterParam
+```
+
+Parameter (in Michelson)
+```
+(pair %configureMinter
+  (address %sender)
+  (pair
+    (address %minter)
+    (nat %tokenId)
+))
+```
+
+- Adds `minter` to global minter list to allow him mint tokens of specific
+  token type.
+
+- Sender must be master minter.
+
+- Minter cannot mint tokens untill `setMintingAllowance`
+  is specified.
+
+**removeMinter**
+
+Types
+```
+tokenId = nat
+
+removeMinter =
+  ( address :sender
+  , address :minter
+  , tokenId :tokenId
+  )
+```
+
+Parameter (in Michelson)
+```
+(pair %removeMinter
+  (address %sender)
+  (pair
+    (address %minter)
+    (nat %tokenId)
+))
+```
+
+- Removes minter from the global minter list for some specific token type
+  and sets its minting allowance to 0. Once minter is removed it will no
+  longer be able to mint or burn tokens.
+
+- Sender must be master minter.
+
+**setMintingAllowance**
```

Comment:
See https://blog.smartdec.net/erc20-approve-issue-in-simple-words-a41aaf47bca6, maybe we should consider splitting it into two functions. Not a showstopper, ofc, but nice to _not_ have this thing mentioned in the audit report in future.

Reply:
IMSM we already had such solution earlier when I proposed to add `approveCAS`  method here 🤷‍♂